### PR TITLE
Preops 1332

### PR DESCRIPTION
**Pull request by ChristinaAdair**
Content from Confluence page has been added to the Tech Note.  I'm not sure about some of the formatting and can use a second opinion about the choices I made.  THANKS

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -6,24 +6,474 @@
 
 .. TODO: Delete the note below before merging new content to the main branch.
 
-.. note::
-
-   **This technote is a work-in-progress.**
+.. Make in-text citations with: :cite:`bibkey`.
```
Comment (plazas) [inline index.rst:9]: Note sure where to put this comment, so I'll place it here at the top: when I built the page, the LSWT logo that displays for me ta the upper left corner says "Large Synoptic Survey Telescope". This should be updated.

File: index.rst [hunk 2]
```diff
@@ -6,24 +6,474 @@
...
+.. .. rubric:: References
+.. 
+.. .. bibliography:: local.bib lsstbib/books.bib lsstbib/lsst.bib lsstbib/lsst-dm.bib lsstbib/refs.bib lsstbib/refs_ads.bib
+..    :style: lsst_aa
 
 Abstract
 ========
 
-Principles, guidelines, conventions, formatting style, and workflow for tutorials developed by the Community Engagement Team (CET).
+A guideline for anyone creating or updating tutorials for the Rubin science community.
```
Comment (plazas) [inline index.rst:19]: Use Vera C. Rubin Observatory the first time it is used in the text (per the naming convention user guide: https://project.lsst.org/documents/name-use-guide)

File: index.rst [hunk 3]
```diff
@@ -6,24 +6,474 @@
...
+This includes all community-facing demonstrations of how to use the Rubin Science Platform or how to analyze LSST data.
```
Comment (plazas) [inline index.rst:20]: Spell out "LSST" the first time it appears in the text.

File: index.rst [hunk 4]
```diff
@@ -6,24 +6,474 @@
...
+Inclusivity
+-----------
+
+Avoid offensive or exclusionary language at all times (e.g., violent or ableist terms).
+Ensure jargon and acronyms are defined.
+
+Level-appropriate
+-----------------
+
+Tutorials should clearly identify and teach to their target audience (beginner, intermediate, or advanced).
```
Comment (plazas) [inline index.rst:36]: Perhaps the three levels of skill should be after a colon and not in parentheses?

File: index.rst [hunk 5]
```diff
@@ -6,24 +6,474 @@
...
+Consumable
+----------
+
+The user should be able to work through the core contents of a tutorial within 30 minutes.
+Additional examples could be included, but should be clearly labeled as extra steps.
+
+Well-documented
+---------------
+
+High quality documentation should be provided with the tutorial, including narrative descriptions, citations, references,
```
Comment (plazas) [inline index.rst:52]: "High-quality" instead of "High quality"?

File: index.rst [hunk 6]
```diff
@@ -6,24 +6,474 @@
...
+Well-documented
+---------------
+
+High quality documentation should be provided with the tutorial, including narrative descriptions, citations, references, 
+and external links to, e.g., package documentation.
+
+Cross-referenced
+----------------
+
+Tutorials should reference an precursor or advanced tutorials that users should consider as pre-requisit or follow-up resources.
```
Comment (plazas) [inline index.rst:58]: Extra "an" in "...an precursor..."?

File: index.rst [hunk 7]
```diff
@@ -6,24 +6,474 @@
...
+
+Properly credited
```
Comment (plazas) [inline index.rst:60]: In the same spirit, should perhaps be a "Contributions" section as well?

File: index.rst [hunk 8]
```diff
@@ -6,24 +6,474 @@
...
+--------
+
+Use the `template notebook <https://github.com/rubin-dp0/cet-dev/blob/main/template.ipynb>`_ in the rubin-dp0 GitHub 
+Organization's ``cet-dev`` repository as a starting point.
+The template contains an example of the header and the mandatory first section described below.
+
+Header
+^^^^^^
+
+The structure of the header is mandatory.
```
Comment (plazas) [inline index.rst:80]: Perhaps these and further instructions would benefit from screenshots of what is being referred to.

File: index.rst [hunk 9]
```diff
@@ -6,24 +6,474 @@
...
+To the right of the logo list the contact author, date last verified, version, container size, and targeted learning level.
+
+The second, third, fourth, and fifth markdown cells should contain a very brief description,
+a list of core skills, a list of the LSST data products, and a list of the python packages used by the notebook.
+List the packages being taught first (e.g., afwDisplay for a notebook about displaying images), and then supporting packages
+(e.g., lsst.daf.butler for a notebook about displaying images).
+It is OK to leave basic support packages (os, glob, numpy, matplotlib) out of this list.
+The contents of cells two through five are used to generate a table of notebook metadata in the README.md file for the repository.
+
+The sixth and seventh markdown cells should contain the credits and acknowledgements, and information about where users go to get support.
```
Comment (plazas) [inline index.rst:92]: "...where users should go..."

File: index.rst [hunk 10]
```diff
@@ -6,24 +6,474 @@
...
+The structure of the header is mandatory.
+
+In the first markdown cell, display the Rubin Observatory logo at upper left.
+To the right of the logo list the contact author, date last verified, version, container size, and targeted learning level.
+
+The second, third, fourth, and fifth markdown cells should contain a very brief description,
+a list of core skills, a list of the LSST data products, and a list of the python packages used by the notebook.
+List the packages being taught first (e.g., afwDisplay for a notebook about displaying images), and then supporting packages
+(e.g., lsst.daf.butler for a notebook about displaying images).
+It is OK to leave basic support packages (os, glob, numpy, matplotlib) out of this list.
```
Comment (plazas) [inline index.rst:89]: "(e.g., os, glob...)"

File: index.rst [hunk 11]
```diff
@@ -6,24 +6,474 @@
...
+^^^^^^
+
+The structure of the header is mandatory.
+
+In the first markdown cell, display the Rubin Observatory logo at upper left.
+To the right of the logo list the contact author, date last verified, version, container size, and targeted learning level.
+
+The second, third, fourth, and fifth markdown cells should contain a very brief description,
+a list of core skills, a list of the LSST data products, and a list of the python packages used by the notebook.
+List the packages being taught first (e.g., afwDisplay for a notebook about displaying images), and then supporting packages
```
Comment (plazas) [inline index.rst:87]: Should code terms such as ``afwDisplay`` and ``lsst.daf.butler`` have a different font or style?

File: index.rst [hunk 12]
```diff
@@ -6,24 +6,474 @@
...
+
+
+
+Tables and plots
+----------------
+
+Table data format
+^^^^^^^^^^^^^^^^^
+
+Results from a TAP service search are best displayed as an astropy table using ``.to_table()``,
```
Comment (plazas) [inline index.rst:133]: Spell out TAP the first time it is used.

File: index.rst [hunk 13]
```diff
@@ -6,24 +6,474 @@
...
+or as a pandas dataframe using ``.to_table().to_pandas()``.
+However, do not use the ``.to_table().show_in_notebook()`` method.
+This can cause issues in the RSP JupyterLab environment that make the notebook hang indefinitely.
+
+Plot color palette
+^^^^^^^^^^^^^^^^^^
+
+To be colorblind-friendly, plots should use the matplotlib color tables viridis or
```
Comment (plazas) [inline index.rst:141]: Should we use a different term from "color blind"? (we'd need to do some research on this)

File: index.rst [hunk 14]
```diff
@@ -6,24 +6,474 @@
...
+`cividis <https://matplotlib.org/stable/users/prev_whats_new/whats_new_2.2.html#cividis-colormap>`_ (or a greyscale),
+or the new `tableau-colorblind10 <https://viscid-hub.github.io/Viscid-docs/docs/dev/styles/tableau-colorblind10.html>`_. 
+
+:: 
+
+  import matplotlib.pyplot as plt
+  plt.style.use('tableau-colorblind10')
+
+
+For the LSST filter set ugrizy, adopt the same colors as DES, which were chosen to be colorblind-friendly:
```
Comment (plazas) [inline index.rst:151]: Spell out DES the first time it is used. Should "ugrizy" be in a different font?

File: index.rst [hunk 15]
```diff
@@ -6,24 +6,474 @@
...
+
+::
+
+  plot_filter_colors = {'u': '#56b4e9', 'g': '#008060', 'r': '#ff4000', 'i': '#850000', 'z': '#6600cc', 'y': '#000000'}
+
+
+Image orientation
+^^^^^^^^^^^^^^^^^
+
+If using a WCS, display east left, north up.
```
Comment (plazas) [inline index.rst:161]: Spell out WCS.

File: index.rst [hunk 16]
```diff
@@ -6,24 +6,474 @@
...
+If only using pixels, (0,0) should be lower left, which is the default for awfDisplay.
+
+When using other plotting packages, transformations might be needed in order to match the afwDisplay default.
+See the LSST Science Pipelines documentation about `Image Indexing <https://pipelines.lsst.io/modules/lsst.afw.image/indexing-conventions.html>`_.
+
+Since use of "extent" is necessry for displaying a WCS overlay for deepCoaads, use it all the time:
```
Comment (plazas) [inline index.rst:167]: typo: "necessry"

Comment (plazas) [reply index.rst:167]: Should "deepCoadss" have a different font/style?

File: index.rst [hunk 17]
```diff
@@ -6,24 +6,474 @@
...
+
+
+Clearing memory
+---------------
+
+These are optional methods for keeping memory use manageable in notebooks which are, e.g., demonstrating
+data visualization techniques with big datasets.
+
+To reduce the memory footprint of a notebook, remove figures once they're no longer needed.
+See the ``remove_figure`` function defined in the DP0 notebook `03_Image_Display_and_Manipulation.ipynb <https://github.com/rubin-dp0/tutorial-notebooks/blob/main/03a_Image_Display_and_Manipulation.ipynb>`__.
```
Comment (plazas) [inline index.rst:187]: Spell out DP0 the first time.

File: index.rst [hunk 18]
```diff
@@ -6,24 +6,474 @@
...
+
+*A method for clearing memory of, for example, large arrays that are not going to be used further on in the notebook is in development.*
```
Comment (plazas) [inline index.rst:189]: This note should be in a box or something similar.

File: index.rst [hunk 19]
```diff
@@ -6,24 +6,474 @@
...
+Consider more pedagogical alternatives when possible (e.g., printing schema columns would also fail if the TAP service was not instantiated).
+
+
+Known warnings
+--------------
+
+If a code cell produces a warning which is known, the preferred method to ignore warnings is to add a markdown cell,
+*before* the code cell which produces the warning, to tell the user it is OK to ignore.
+
+*The appropriate use-cases for ignoring categories of warnings (below) is still under consideration.*
```
Comment (plazas) [inline index.rst:210]: This note should be in a box or something similar (with a warning).

File: index.rst [hunk 20]
```diff
@@ -6,24 +6,474 @@
...
+This is not preferred because ignore categories of warnings can allow real issues to go unnoticed.
```
Comment (plazas) [inline index.rst:211]: If this is not preferred behavior, there should be a more visual warning. Or we should reconsider placing it here, as we don't want to encourage bad habits.

File: index.rst [hunk 21]
```diff
@@ -6,24 +6,474 @@
...
+------------------
+
+Avoid using comments within a code cell as documentation.
+Markdown cells are the preferred way to provide descriptive text.
+
+
+Code cell style standard PEP8
+-----------------------------
+
+``PEP8`` is the standard, and ``flake8`` is a tool to ensure compliance with the standards.
```
Comment (plazas) [inline index.rst:239]: Perhaps place a link with a reference to what PEP8 is?

File: index.rst [hunk 22]
```diff
@@ -6,24 +6,474 @@
...
+Use ``flake8`` to ensure notebook code conforms to codebase style `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_ , with a few exceptions. 
+
+Notebook tutorial developers must install the following packages locally in their home directory:
+
+::
+
+  pip install --user flake8-nb
+  pip install --user pycodestyle_magic
+
+It is known that the most up-to-date version of flake8 has some issues.
```
Comment (plazas) [inline index.rst:250]: ``flake8``

File: index.rst [hunk 23]
```diff
@@ -6,24 +6,474 @@
...
+::
+
+  flake8-nb notebook_name.ipynb
+
+
+This will give you a final check of any violations with ``PEP8``.
+This will catch things that can be missed line-by-line, such as packages that are imported but never used.  
+
+
+Git branch, merge, and review policy for ``tutorial-notebooks``
```
Comment (plazas) [inline index.rst:311]: If possible, in the instructions below, we should write down the explicit git commands we'd like people to use (as in the DM Developers Guide).

File: index.rst [hunk 24]
```diff
@@ -6,24 +6,474 @@
...
+
+Pull request
+------------
+
+When the notebook is complete open a pull request to merge the ticket branch into the ``main`` branch (*not* to ``prod``). 
+
+Review
+------
+
+Contact one or more Rubin staff members (it need not be a Community Engagement team member) with the appropriate expertise
```
Comment (plazas) [inline index.rst:341]: Where could people ask for reviewers if they don't know anyone a priori? Perhaps in "Community"?

File: index.rst [hunk 25]
```diff
@@ -6,24 +6,474 @@
...
+Ensure that all of the reviewers comments are addressed.
+Make changes and new commits to the branch, and respond to all of their comments with either a confirmation a change was made,
+or an explanation of why the request was not implemented. 
+
+Contact the reviewers to let them know the pull request now awaits their approval.
+
+Merge
+-----
+
+After the reviewers have approved the pull request, rebase and merge your ticket branch into the ``main`` branch (*not* to prod).
```
Comment (plazas) [inline index.rst:354]: In the HTML version, there is a space missing between "not" and "to prob".

Comment (plazas) [reply index.rst:354]: Prod should have different formatting: ``prod``

File: index.rst [hunk 26]
```diff
@@ -6,24 +6,474 @@
...
+Remember to make comments in the associated Jira tickets about the major updates and mark the ticket as done.
+
+
+Updates to the RSP's recommended version
+----------------------------------------
+
+Decisions on whether to bump the recommended image for the RSP are made jointly between the CET and the RSP team.
+Once the decision has been made, a PREOPS ticket will be created and assigned to a CET member.
+
+Bumping the recommended always occurs during a Patch Thursday window.
```
Comment (plazas) [inline index.rst:381]: What is a "Patch Thursday window"?

File: index.rst [hunk 27]
```diff
@@ -6,24 +6,474 @@
...
+Remember to set the PREOPS ticket to done.
+
+
+Portal tutorial format and style
+================================
+
+The portal tutorials are written in RST format and are kept within the data release documentation at
+`dp0-2.lsst.io/tutorials-examples <https://dp0-2.lsst.io/tutorials-examples/index.html>`__.
+
+All portal tutorials should have a descriptive title, and list the contact authors, the date last verified to run, and the targeted learning level.
```
Comment (plazas) [inline index.rst:407]: And the RSP/code version that was last used to run?

File: index.rst [hunk 28]
```diff
@@ -6,24 +6,474 @@
...
+
+During the Patch Thursday window, after the recommended has been bumped, release to ``prod`` following the steps above.
+
+Remember to set the PREOPS ticket to done.
+
+
+Portal tutorial format and style
+================================
+
+The portal tutorials are written in RST format and are kept within the data release documentation at
```
Comment (plazas) [inline index.rst:404]: Spell our RST

File: index.rst [hunk 29]
```diff
@@ -6,24 +6,474 @@
...
+`dp0-2.lsst.io/tutorials-examples <https://dp0-2.lsst.io/tutorials-examples/index.html>`__.
+
+All portal tutorials should have a descriptive title, and list the contact authors, the date last verified to run, and the targeted learning level.
+A brief narrative introduction to the tutorial should be provided at the top of the page.
+
+The rest of the portal tutorial should be divided into sequentially numbered steps and sub-steps.
+Use descriptive text and screenshots to demonstrate what the user should do.
+Augment screenshots with arrows or circles to guide the users attention as needed.
+
+Ensure that any ADQL is put into code boxes in RST so that users may copy-and-paste whenever possible.
```
Comment (plazas) [inline index.rst:414]: Spell our ADQL the first time it's used.

File: index.rst [hunk 30]
```diff
@@ -6,24 +6,474 @@
...
+
+Accessibility
+-------------
+
+Improve tutorials' accessibility to people with visual disabilities by finding and implementing, e.g., screen reader compatibility software, data sonification packages.
+
+Translations
+------------
+
+At minimum, translate any undergraduate-level tutorials into Spanish.
```
Comment (plazas) [inline index.rst:460]: Can we offer tools or some sort of help for people to do this? For example, should Rubin buy a tool (e.g., deepl.com) or pay people to do it?

File: index.rst [hunk 31]
```diff
@@ -6,24 +6,474 @@
...
+
+Narrative voice
+---------------
+
+Adopt a best practice for narrative instructions: should we use terms like "you" or "let's"? Or should a more passive voice be adopted?
+
+Accessibility
+-------------
+
+Improve tutorials' accessibility to people with visual disabilities by finding and implementing, e.g., screen reader compatibility software, data sonification packages.
```
Comment (plazas) [inline index.rst:455]: We should provide examples/references of how to do this.  Maybe we can do this with one or a few of the existing tutorials.

**Review by plazas — CHANGES_REQUESTED** [review]
Thanks for this! I left a few comments; I'd be happy to take a look again later. Also, should this branch be called "tickets/PREOPS-1332" instead of just "PREOPS-1332"?

File: index.rst (hunk 2 above)
Comment (ChristinaAdair) [reply index.rst:19]: Good catch, made a branch for tickets/PREOPS-1332, from PREOPS-1332.  Also added Vera C. Rubin

File: index.rst (hunk 3 above)
Comment (ChristinaAdair) [reply index.rst:20]: Done

File: index.rst (hunk 4 above)
Comment (ChristinaAdair) [reply index.rst:36]: Done

File: index.rst (hunk 5 above)
Comment (ChristinaAdair) [reply index.rst:52]: Done

File: index.rst (hunk 6 above)
Comment (ChristinaAdair) [reply index.rst:58]: done, also corrected "pre-requisit" to "prerequisite"

File: index.rst (hunk 9 above)
Comment (ChristinaAdair) [reply index.rst:92]: done

File: index.rst (hunk 10 above)
Comment (ChristinaAdair) [reply index.rst:89]: done

File: index.rst (hunk 11 above)
Comment (ChristinaAdair) [reply index.rst:87]: done

File: index.rst (hunk 12 above)
Comment (ChristinaAdair) [reply index.rst:133]: done

File: index.rst (hunk 15 above)
Comment (ChristinaAdair) [reply index.rst:161]: Done

File: index.rst (hunk 17 above)
Comment (ChristinaAdair) [reply index.rst:187]: done

File: index.rst (hunk 22 above)
Comment (ChristinaAdair) [reply index.rst:250]: done

File: index.rst (hunk 25 above)
Comment (ChristinaAdair) [reply index.rst:354]: I think there is a space between not and "to prod", also changed prod to ``prod``

File: index.rst (hunk 27 above)
Comment (ChristinaAdair) [reply index.rst:407]: added

File: index.rst (hunk 1 above)
Comment (ChristinaAdair) [reply index.rst:9]: Square team is updating the template for distribution

File: index.rst (hunk 7 above)
Comment (ChristinaAdair) [reply index.rst:61]: Do you mean that we need a section devoted to "Contributions" in the header of a tutorial?

File: index.rst (hunk 8 above)
Comment (ChristinaAdair) [reply index.rst:80]: Going to check with Melissa on this comment and add to the next version.

File: index.rst (hunk 13 above)
Comment (ChristinaAdair) [reply index.rst]: Going to leave for now

File: index.rst (hunk 14 above)
Comment (ChristinaAdair) [reply index.rst]: done

File: index.rst (hunk 16 above)
Comment (ChristinaAdair) [reply index.rst]: done

File: index.rst (hunk 18 above)
Comment (ChristinaAdair) [reply index.rst]: done

File: index.rst (hunk 19 above)
Comment (ChristinaAdair) [reply index.rst]: done

File: index.rst (hunk 20 above)
Comment (ChristinaAdair) [reply index.rst]: Added into the note

File: index.rst (hunk 21 above)
Comment (ChristinaAdair) [reply index.rst:251]: Changed the name of the link text

File: index.rst (hunk 23 above)
Comment (ChristinaAdair) [reply index.rst]: At this time, we want to be flexible to the tools people are using (e.g. web UI v. command line)

File: index.rst (hunk 24 above)
Comment (ChristinaAdair) [reply index.rst]: Added link and instructions to post on community.lsst.org

File: index.rst (hunk 28 above)
Comment (ChristinaAdair) [reply index.rst]: done

File: index.rst (hunk 26 above)
Comment (ChristinaAdair) [reply index.rst]: Added language to clarify

File: index.rst [hunk 32]
```diff
@@ -6,24 +6,498 @@
...
-.. Make in-text citations with: :cite:`bibkey`.
-.. Uncomment to use citations
-.. .. rubric:: References
-.. 
-.. .. bibliography:: local.bib lsstbib/books.bib lsstbib/lsst.bib lsstbib/lsst-dm.bib lsstbib/refs.bib lsstbib/refs_ads.bib
-..    :style: lsst_aa
+Inclusivity
+-----------
+
+Offensive or exclusionary language is never permitted (e.g., violent or ableist terms).
```
Comment (jeffcarlin) [inline index.rst:30]: This is rather terse - is there a code of conduct or something similar that we could link to? (Either a DP0-specific statement, or the "standard" Rubin or NOIRLab statements?)

File: index.rst (hunk 31 above)
Comment (ChristinaAdair) [reply index.rst:482]: Part of our stretch goals, discuss at CET meeting

File: index.rst [hunk 33]
```diff
@@ -6,24 +6,498 @@
...
+
+Known warnings
+--------------
+
+If a code cell produces a warning which is known and it should be ignored, the preferred method is to add a markdown cell
+*before* the code cell which produces the warning, to tell the user it is acceptable to ignore.
+
+.. Note::
+
+   This is not preferred because ignore categories of warnings can allow real issues to go unnoticed.
```
Comment (jeffcarlin) [inline index.rst:220]: "ignore" --> "ignoring"

File: index.rst [hunk 34]
```diff
@@ -6,24 +6,498 @@
...
+
+Commit and push
+---------------
+
+Always restart the kernel and clear all outputs before saving, committing, and pushing changes.
+
+Pull request
+------------
+
+When the notebook is complete open a pull request to merge the ticket branch into the ``main`` branch (again, *note* to ``prod``).
```
Comment (jeffcarlin) [inline index.rst]: "note" --> "not"?

File: index.rst [hunk 35]
```diff
@@ -6,24 +6,498 @@
...
+Ensure that all of the reviewers' comments are addressed.
+Make changes and new commits to the branch, and respond to all of their comments with either a confirmation a change was made,
+or an explanation of why the request was not implemented.
+
+Contact the reviewers to let them know the pull request now awaits their approval.
+
+Merge
+-----
+
+After the reviewers have approved the pull request, ``rebase and merge`` your ticket branch into the ``main`` branch (again, *note* to ``prod``).
```
Comment (jeffcarlin) [inline index.rst:369]: "note" --> "not"?

File: index.rst [hunk 36]
```diff
@@ -6,24 +6,498 @@
...
+
+Notebook metadata
+-----------------
+
+Embed notebook metadata (e.g., version, skills, packages) in a way that can be scraped and used to auto-generate the ``README.md`` file or a Table of Contents, to enable users to browse notebook contents.
+
+Narrative voice
+---------------
+
+Adopt a best practice for narrative instructions: should we use terms like "you," or should a passive voice be adopted?
```
Comment (jeffcarlin) [inline index.rst:478]: I vote against passive voice, 'cause I generally am not a fan...

File: index.rst [hunk 37]
```diff
@@ -6,24 +6,498 @@
...
+
+At minimum, translate any undergraduate-level tutorials into Spanish.
+
+Additionally, improve tutorials' accessibility to non-English speakers by finding and implementing automatic translation and localization software.
+
+Purge extraneous items in notebooks
+-----------------------------------
+
+Develop a best practice for how to keep notebook memory usage in check, in addition to deleting figures.
+Do not rely on the ``del`` command for this.
```
Comment (jeffcarlin) [inline index.rst:496]: Should we explain why that's not sufficient? (I actually don't know!)

**Comment by ChristinaAdair** [comment]
Renamed current branch to tickets/PREOPS-1332, included all merges and comments up to this point into tickets/PREOPS-1332. A new pull request will be created with the reviewers

File: index.rst [hunk 38]
```diff
@@ -6,24 +6,498 @@
...
+
+Recipe functions
+----------------
+
+Create recipes for common user activities.
+These could be, e.g., ADQL searches for the portal, code snippets for the command line, or python modules that can be imported.
```
Comment (jeffcarlin) [inline index.rst:502]: If we develop python modules, then we'll need to figure out how/where to make those available?

**Review by jeffcarlin — APPROVED** [review]

**Pull request closed without merging**